agent.go

```diff
@@ -33,8 +33,9 @@ type (
 
 	// EdgeStackConfig represent an Edge stack config
 	EdgeStackConfig struct {
-		Name        string
-		FileContent string
+		Name         string
+		FileContent  string
+		ImageMapping map[string]string // a map of stackfile image to imageCache url(with sha)
```

Comment:
I'd try to rename the attribute so it's clear what it is without a comment, makes it easier to work with. Does something like this make sense?
```suggestion
		StackImagesURLs map[string]string
```